mobile/src/main/java/ch/epfl/sdp/mobile/ui/tournaments/PoolTable.kt

```diff
@@ -0,0 +1,380 @@
+package ch.epfl.sdp.mobile.ui.tournaments
+
+import androidx.compose.animation.*
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.*
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.Stable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.draw.drawBehind
+import androidx.compose.ui.draw.rotate
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.Layout
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.style.TextOverflow
+import androidx.compose.ui.unit.*
+import androidx.compose.ui.util.fastForEach
+import androidx.compose.ui.util.fastForEachIndexed
+import androidx.compose.ui.util.fastMap
+import ch.epfl.sdp.mobile.state.LocalLocalizedStrings
+import ch.epfl.sdp.mobile.ui.FadeEnterTransition
+import ch.epfl.sdp.mobile.ui.FadeExitTransition
+import ch.epfl.sdp.mobile.ui.PawniesColors
+import kotlin.math.roundToInt
+
+/** Represents an amount of points from a given player. */
+typealias PoolScore = Int
+
+/** A pool member is a participant of the pool. */
+@Stable
+interface PoolMember {
+
+  /** The name of the participant. */
+  val name: String
+
+  /** The total score of the participant, or null if the pool has not started yet. */
+  val total: PoolScore?
+}
+
+/**
+ * The information displayed within the pool.
+ *
+ * @param P the type of the actual [PoolMember]s.
+ */
+@Stable
+interface PoolData<P : PoolMember> {
+
+  /** Returns a [List] of all the participants to this pool. */
+  val members: List<P>
+
+  /**
+   * Returns the score of [this] participant against an [other] participant, or null if the pool has
+   * not started yet.
+   *
+   * @receiver the current participant.
+   * @param other the other participant.
+   * @return the score, or null if the pool has not started yet.
+   */
+  fun P.scoreAgainst(other: P): PoolScore?
+}
+
+/**
+ * The table of the players which are participating in a pool.
+ *
+ * @param P the type of the [PoolMember] which participate in the pool.
+ * @param data the [PoolData] which must be displayed.
+ * @param modifier the [Modifier] for this composable.
+ */
```

Comment:
Missing parameter documentation
```suggestion
 * @param textStyle the [TextStyle] for this composable.
 */
```